driver/outer_driver.hpp

```diff
@@ -0,0 +1,517 @@
+
+    size_t cnt = 0;
+    for(size_t i = 0; i < in_n; i++)
+    {
+        for(size_t j = 0; j < in_m; j++)
+        {
+            outputhost[cnt]   = 0;
+            outputhost[cnt++] = input1[i] * input2[j];
+        }
+    }
+    return ret;
+}
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloSumBackwardRunHost(miopenTensorDescriptor_t input1Desc,
+                              miopenTensorDescriptor_t input2Desc,
+                              miopenTensorDescriptor_t input1GradDesc,
+                              miopenTensorDescriptor_t input2GradDesc,
+                              miopenTensorDescriptor_t yGradDesc,
+                              Tgpu* input1,
+                              Tgpu* input2,
+                              Tgpu* outGrad,
+                              Tcheck* in1Gradhost,
+                              Tcheck* in2Gradhost)
+{
+    auto input1_dims = miopen::deref(input1Desc).GetLengths();
+    auto input2_dims = miopen::deref(input2Desc).GetLengths();
+    auto output_dims = miopen::deref(yGradDesc).GetLengths();
+
+    size_t in_n = input1_dims[0];
+    size_t in_m = input2_dims[0];
+
+    int32_t ret = 0;
+
+    for(size_t i = 0; i < in_n; i++)
+    {
+        Tcheck sum = static_cast<Tcheck>(0.0f);
+        for(size_t j = 0; j < in_m; j++)
+        {
+            sum += static_cast<Tcheck>(input2[j]) * static_cast<Tcheck>(outGrad[i * in_m + j]);
+        }
+        in1Gradhost[i] = sum;
+    }
+
+    for(size_t j = 0; j < in_m; j++)
+    {
+        Tcheck sum = static_cast<Tcheck>(0.0f);
+        for(size_t i = 0; i < in_n; i++)
+        {
+            sum += static_cast<Tcheck>(input1[i]) * static_cast<Tcheck>(outGrad[i * in_m + j]);
+        }
+        in2Gradhost[j] = sum;
+    }
+    return ret;
+}
+
+template <typename Tgpu, typename Tref>
+class OuterDriver : public Driver
+{
+public:
+    OuterDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&input1Desc);
+        miopenCreateTensorDescriptor(&input2Desc);
+        miopenCreateTensorDescriptor(&yDesc);
+
+        miopenCreateTensorDescriptor(&input1GradDesc);
+        miopenCreateTensorDescriptor(&input2GradDesc);
+        miopenCreateTensorDescriptor(&yGradDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    std::vector<int> GetInputTensorLengthsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~OuterDriver() override
+    {
+        miopenDestroyTensorDescriptor(input1Desc);
+        miopenDestroyTensorDescriptor(input1Desc);
+        miopenDestroyTensorDescriptor(yDesc);
```

Comment:
Missing TensorDesciptors for bwd?

Reply:
I add "miopenDestroyTensorDescriptor" operations for TensorDesciptors for bwd.